### ISSUE TITLE
你好我遇到了这个问题：tree file mnistof/rqalsh/0.rqalsh does not exist

### ISSUE THREAD

**Issue by ghost** [HuangQiang/RQALSH#1]

**Comment by ghost** [comment]
ftp_18@L0-2:~/RQALSH/RQALSH-master$ ./rqalsh -alg 3 -n 69000 -d 784 -B 4 -beta 100 -delta 0.49 -c 2 -ds ./mnistD.txt -df mnistdf -of mnistof
alg           = 3
n             = 69000
d             = 784
B             = 4
beta          = 100
delta         = 0.49
c             = 2.0
data_set      = ./mnistD.txt
data_folder   = mnistdf
output_folder = mnistof

Read Data: 6.557659 Seconds

Write Dataset in New Format: 0.000023 Seconds

Hash tables exist.
ftp_18@L0-2:~/RQALSH/RQALSH-master$ ./rqalsh -alg 4 -qn 1000 -d 784 -qs mnistQ.txt -ts gt.fn2.0 -df mnistdf -of mnistof
alg           = 4
qn            = 1000
d             = 784
query_set     = mnistQ.txt
truth_set     = gt.fn2.0
data_folder   = mnistdf
output_folder = mnistof

Read Query and Ground Truth: 0.134205 Seconds

tree file mnistof/rqalsh/0.rqalsh does not exist

**Comment by HuangQiang** [comment]
For -alg 3, in this package, we use the file "para" to check whether the hash tables exist or not (pls check rqalsh.cc, function: RQALSH::write_params() ). As the program output "Hash tables exist", the program did NOT write hash tables to disk.  Thus, for -alg 4, the tree file 0.rqalsh does not exist. 

To fix this problem, please ensure you remove all files in the index file before you build new index. i.e., remove all file in folder "mnistof/rqalsh/" in your case. You could remove the "para" file in the folder "mnistof/rqalsh/" and re-run -alg 3 and -alg -4, then you will find the hash tables in the folder "mnistof/rqalsh/" and get the correct results. 

Please leave a message if you have any other problems. I will collect the comments and polish the source codes in next version.

**Comment by ghost** [comment]
你好，我的问题解决了我把B那个参数给错了，谢谢

**Comment by HuangQiang** [comment]
yeah, you should set a larger value of B either. I suggest you set B to be 16384 (16 KB).
